examples/README.md

```diff
@@ -0,0 +1,233 @@
+Run_time_dispatching will be build if `-DBUILD_EXAMPLES=true` and `-DBUILD_SHARED_LIBS=true`
+
+The example executable naming convention follows `example_<$domain>_<$routine>_<$backend>` for compile-time dispatching examples 
+  or `example_<$domain>_<$routine>` for run-time dispatching examples. 
+  E.g. `example_blas_gemm_usm_mklcpu_cublas `  `example_blas_gemm_usm`
+
+## Test outputs (blas, rng, lapack)
+  
+## blas
+
+Run-time dispatching examples with mklcpu backend
+```
+$ export SYCL_DEVICE_FILTER=cpu
+$ ./bin/example_blas_gemm_usm
+
+########################################################################
+# General Matrix-Matrix Multiplication using Unified Shared Memory Example:
+#
+# C = alpha * A * B + beta * C
+#
+# where A, B and C are general dense matrices and alpha, beta are
+# floating point type precision scalars.
+#
+# Using apis:
+#   gemm
+#
+# Using single precision (float) data type
+#
+# Device will be selected during runtime.
+# The environment variable SYCL_DEVICE_FILTER can be used to specify
+# SYCL device
+#
+########################################################################
+
+Running BLAS GEMM USM example on CPU device.
+Device name is: Intel(R) Core(TM) i7-6770HQ CPU @ 2.60GHz
+Running with single precision real data type:
+
+                GEMM parameters:
+                        transA = trans, transB = nontrans
+                        m = 45, n = 98, k = 67
+                        lda = 103, ldB = 105, ldC = 106
+                        alpha = 2, beta = 3
+
+                Outputting 2x2 block of A,B,C matrices:
+
+                        A = [ 0.340188, 0.260249, ...
+                            [ -0.105617, 0.0125354, ...
+                            [ ...
+
+
+                        B = [ -0.326421, -0.192968, ...
+                            [ 0.363891, 0.251295, ...
+                            [ ...
+
+
+                        C = [ 0.00698781, 0.525862, ...
+                            [ 0.585167, 1.59017, ...
+                            [ ...
+
+BLAS GEMM USM example ran OK.
+
+```
+
+Compile-time dispatching example with both mklcpu and cublas backend
+```
+./bin/example_blas_gemm_usm_mklcpu_cublas
+
+########################################################################
+# General Matrix-Matrix Multiplication using Unified Shared Memory Example:
+#
+# C = alpha * A * B + beta * C
+#
+# where A, B and C are general dense matrices and alpha, beta are
+# floating point type precision scalars.
+#
+# Using apis:
+#   gemm
+#
+# Using single precision (float) data type
+#
+# Running on both Intel CPU and Nvidia GPU devices
+#
+########################################################################
+
+Running BLAS GEMM USM example
+Running with single precision real data type on:
+        CPU device: Intel(R) Core(TM) i9-7920X CPU @ 2.90GHz
+        GPU device: TITAN RTX
+BLAS GEMM USM example ran OK: CPU and GPU results match
```

Comment:
Ctest will still pass if results from CPU and GPU do not match because we do not exit with error code. Instead the result will show through std::cout or std::cerr. So in ctest we do not check for correctness.
This check function is just to show how users can use results from CPU and GPU (like comparing them

Reply:
I understand, but when you compare results, the expectation is, the result of comparison will be used for correctness. And ctest or gtest does not use that. I suggest either fail in ctest if results do not match, or don't compare the results.

Reply:
Makes sense. Changed to printing result matrices instead of comparing them.